source/manual/intro-to-govuk-docker.html.md

```diff
@@ -0,0 +1,180 @@
+---
+title: Intro to Docker and govuk-docker
+section: Docker
+layout: manual_layout
+parent: "/manual.html"
+last_reviewed_on: 2019-07-01
+review_in: 6 months
+owner_slack: "#govuk-developers"
+---
+
+## Intro
+If you are new to Docker here's a quick intro in the context of the GOV.UK stack.
+
+
+### Install Docker
+
+*   Download Docker [https://www.docker.com/get-started](https://www.docker.com/get-started)
+*   Docker for Mac intro [https://docs.docker.com/docker-for-mac/](https://docs.docker.com/docker-for-mac/)
+
+### Video explaining the concepts in Docker
+
+[https://www.youtube.com/watch?v=i7yoXqlg48M](https://www.youtube.com/watch?v=i7yoXqlg48M) - first 30 minutes
+
+## Learning docker with content-publisher
+We'll run a few things here to get the [content-publisher] up and running in Docker. This is a convoluted example but it will help to explain the concepts involved.
+
+### Run a container from an image
+
+* Make sure you have docker running:
+
+```shell
+$mac docker --version
+Docker version 18.09.2, build 6247962
+```
+
+* run from the root of the [content-publisher] project:
+
+```shell
+$mac cd ~/govuk/content-publisher
+$mac docker run -it --rm ubuntu
+```
+
+##### What are the flags?
+
+```shell
+$mac docker run --help
+...
+-i, --interactive                    Keep STDIN open even if not attached
+...
+-t, --tty                            Allocate a pseudo-TTY
+...
+    --rm                             Automatically remove the container when it exits
+```
+
+#### What are Docker images and a container?
+The `ubuntu` is the "image" that we specify. Using object oriented programming as an analogy, an image is like a "class". When we "instantiate" an instance of a class, these are called "containers".
+
+Images are downloaded from the Docker Registry.
+
+Another image is ruby:
+
+```shell
+docker run -it --rm ruby:2.6.3 bash
+```
+
+Here we get a ruby image and specifying which version we want. The `bash` argument at the end is passed to the docker run command. It will execute this inside the container, which means we get a bash shell.
+
+### Volumes
+
+* Mount a volume by running:
```

Comment:
```suggestion
Mount a volume by running:
```

Reply:
Same thing about bullet points as a step

Reply:
@chao-xian I find it a bit more difficult to read when mixing bullet points with code and other comments that span full width (not indented), but that's just my personal preference 🤷‍♀